robottelo/host_helpers/api_factory.py

```diff
@@ -630,3 +632,129 @@ def template_update(self, temp):
         if not template.locked:
             template.locked = True
             template.update(['locked'])
+
+    def attach_custom_product_subscription(self, prod_name=None, host_name=None):
+        """Attach custom product subscription to client host
+        :param str prod_name: custom product name
+        :param str host_name: client host name
+        """
+        host = self._satellite.api.Host().search(query={'search': f'{host_name}'})[0]
+        product_subscription = self._satellite.api.Subscription().search(
+            query={'search': f'name={prod_name}'}
+        )[0]
+        self._satellite.api.HostSubscription(host=host.id).add_subscriptions(
+            data={'subscriptions': [{'id': product_subscription.id, 'quantity': 1}]}
+        )
+
+    def wait_for_errata_applicability_task(
```

Comment:
Last this was used in some katello-agent tests, which we moved and removed use of this in https://github.com/SatelliteQE/robottelo/pull/9005

Reply:
Keeping this for future need if any same as above function. Do you think due to depreciation of katello-agent tool will also make this definition absolute and cant be fit into what is to be used in errata applicability task ?

Reply:
Yes, katello-agent will be removed completely in coming release and iirc this pulp task was removed in 6.11 for errata applicability so we removed the use of this helper at that time